src/main/groovy/io/openliberty/tools/gradle/tasks/AbstractServerTask.groovy

```diff
@@ -731,7 +787,32 @@ abstract class AbstractServerTask extends AbstractTask {
         }
     }
 
-    private void writeServerEnvProperties(File file, Properties properties, Properties projectProperties) throws IOException {
+    private Properties convertServerEnvToProperties(File serverEnv) {
+        if(serverEnv == null) {
+            return null;
```

Comment:
I think this is a bad idea to return null. Also, perhaps you should check that `if ((serverEnv == null) || !serverEnv.exists())`. I would return the empty Properties object that you are creating at line 796 in this situation.